### PR TITLE
Added nifti script to bin & nifi files to data.

### PR DESCRIPTION
**Pull request by Bronte-Mckeown**
Added script for creating niftis to /bin for each of the CAPs and saved niftis to /data.

### CODE REVIEW

File: bin/cap_to_nifti.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+# get data from combined atlas
+com_atlas = combined.get_fdata()
+
+# store unique numbers from com_atlas as com_parcels
+com_parcels = (np.unique(com_atlas))
+# remove first row as it is zero
+com_parcels = np.delete(com_parcels, (0), axis=0)
+# remove rows after 1054 as we don't have corresponding CAP data 
+com_parcels = np.delete(com_parcels, slice(1054,1080), axis =  0)
```
Comment (htwangtw) [inline bin/cap_to_nifti.py:27]: Okay this is not right. i will need to check the template I created.

Comment (htwangtw) [reply bin/cap_to_nifti.py:27]: I have fixed the bug wrt the atlas - can you update your branch and use the new file, and update your script accordingly?
This is how you update your branch:
```
git checkout cap-to-nifti
git remote add upstream https://github.com/htwangtw/nkicap.git
git fetch upstream
git merge upstream/main
```

File: bin/cap_to_nifti.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        # retrieve cap value
+        current_number = caps[cap].loc[parcel] 
+        # replace mask values with cap value
+        new_nifti_data[mask] = current_number
+
+    # for each cap, plot nifti and save to results
+    plotting.plot_stat_map(nb.Nifti1Image(new_nifti_data, header=combined.header, affine=combined.affine), title="Combined_{}".format(cap))
+    plotting.show()
+    nifti = nb.Nifti1Image(new_nifti_data, header=combined.header, affine=combined.affine)
+    nb.save(nifti, 'data/combined_atlas_{}.nii.gz'.format(cap))
```
Comment (htwangtw) [inline bin/cap_to_nifti.py:49]: Can you save the nifti files to `results/cap_nifti/group_cap<number>.nii.gz`?
I would prefer to have nothing but outputs from `make_dataset.py` and the files on the git repo?

**Review by htwangtw — CHANGES_REQUESTED** [review]
Thanks for submitting the PR. This also helped me to find a bug in the atlas combining script. (Note to self: write your freaking tests.)

File: bin/cap_to_nifti.py (hunk 1 above)
Comment (Bronte-Mckeown) [reply bin/cap_to_nifti.py]: Great, thanks for fixing. I've updated my branch, used the new file and updated the script accordingly.

File: bin/cap_to_nifti.py (hunk 2 above)
Comment (Bronte-Mckeown) [reply bin/cap_to_nifti.py]: I have changed this now so that the niftis are saved to that folder.

**Review by htwangtw — APPROVED** [review]
Thank you this looks good! 
Can you also commit the new nifti files create?
May I ask to rename the files from `group_{}.nii.gz` to `group_cap_{}.nii.gz`?
The current name is not clear on its own.

Also, to make PEP8 complain less, in the repo run `poetry black .` before committing so it automatically format the code style.

**Comment by Bronte-Mckeown** [comment]
> Thank you this looks good!
> Can you also commit the new nifti files create?
I think I have done that now!
> May I ask to rename the files from `group_{}.nii.gz` to `group_cap_{}.nii.gz`?
I think they already were as I used the column names (e.g., cap_01) to name them? But please let me know if you meant something else.
> Also, to make PEP8 complain less, in the repo run `poetry black .` before committing so it automatically format the code style.
Ok I will make sure to try this next time I update a script!

**Comment by htwangtw** [comment]
> > Thank you this looks good!
> > Can you also commit the new nifti files create?
> > I think I have done that now!
> > May I ask to rename the files from `group_{}.nii.gz` to `group_cap_{}.nii.gz`?
> > I think they already were as I used the column names (e.g., cap_01) to name them? But please let me know if you meant something else.

You are right - revert my changes :bowing_woman:

**Comment by Bronte-Mckeown** [comment]
Please bare with me, I'm still a bit of a github noob- should I simply delete the cap-to-nifti branch from my fork now that it's been merged to your main and I've pulled upstream?

**Comment by htwangtw** [comment]
Yes you can delete it - no harm keeping it either. It's up to you!

**Pull request merged**